### ISSUE TITLE
Inconsistent wxArtProvider bitmaps

### ISSUE THREAD

**Issue by rtrahan504** [wxWidgets/wxWidgets#22095]
![image](https://user-images.githubusercontent.com/23139439/152581663-43a0b9d9-6c2b-40d9-964b-c3dc77c106bc.png)

```
m_menuItem_NewConfig->SetBitmap(wxArtProvider::GetBitmap(wxART_NEW, wxART_MENU, wxSize(16, 16)));
m_menuItem_CutConfig->SetBitmap(wxArtProvider::GetBitmap(wxART_CUT, wxART_MENU, wxSize(16, 16)));
m_menuItem_CopyConfig->SetBitmap(wxArtProvider::GetBitmap(wxART_COPY, wxART_MENU, wxSize(16, 16)));
m_menuItem_PasteConfig->SetBitmap(wxArtProvider::GetBitmap(wxART_PASTE, wxART_MENU, wxSize(16, 16)));
m_menuItem_DeleteConfig->SetBitmap(wxArtProvider::GetBitmap(wxART_DELETE, wxART_MENU, wxSize(16, 16)));
```

Latest stable release on windows. Reproducible in many situations.

**Comment by vadz** [comment]
Is this with DPI scaling or does it happen even at 100% DPI?

**Comment by rtrahan504** [comment]
I have not manually changed anything in wx with regards to DPI (not sure if that's even possible). One of my monitors is 4k so windows 10 has the "Scale & Layout" setting in the Display settings set to 150% by default on that monitor. Other monitors are just HD so scaling is at 100%. Regardless of these settings, each bitmap from the art provider is being sized _inconsistently_. If the sizes were consistently wrong that's one issue, but the inconsistency is hard to work with. 

Also, it's not easy to consistently reproduce the issue since it is inherently inconsistent for all bitmaps from the art provider. The same wxArtId may get sized differently when retrieved from the art provider. Whether the bitmap from the art provider is put into a wxbutton, wxmenuitem, wxstaticbitmap, etc also affects the inconsistency.

Very similar bugs have been reported back to 2015 and closed but not resolved. A google search shows similar issues on this repo being closed several times. For example #17071. This bug essentially renders the art provider useless for production code since it looks so horrible and can affect widget layout.

**Comment by vadz** [comment]
Are using 3.0? #17071 was only fixed in 3.1, and it does indeed look like the same issue.

**Comment by rtrahan504** [comment]
Ah. Apologies. I didn't realize issues from 2015 were already on 3.1 and not the stable release 3.0 from 2020. I'm stuck using the stable release.
